src/lib/TypeWriter.svelte

```diff
@@ -59,13 +65,20 @@
 		if (!textArr.length) return
 
 		for (let i = 0; iterations === 0 || i < iterations; i++) {
-			for (const text of textArr) {
+			for (let j = 0; j < textArr.length; j++) {
+				// Callback
+				onTextStart(j)
+
+				const text = textArr[j]
```

Comment:
Probably no real difference, but I would change the order here
```suggestion
				const text = textArr[j]

				ontypestart?.(j)
```